parser.js

```diff
@@ -177,6 +185,12 @@ exports.version = require("./package.json").version;
 
 exports.parse = parse;
 
+exports.parseForESLint = function parseForESLint(code, options) {
+    options.parseForESLint = true;
```

Comment:
I think it would be best to avoid mutating the options object provided by the user. (This was also a problem in https://github.com/eslint/espree/issues/329.)

Reply:
In which case I would be forced to change the signature of `parse()` to accept an optional third argument. There's no reason that would be an issue, right?

Reply:
I don't think it would be an issue.

If you don't want the change to be visible to the public API, you could create a helper function, e.g. `parseMaybeForESlint(code, options, internalOptions)`, and have `parse` call `parseMaybeForESLint(code, options, { isForESLint: false })` and have `parseForESLint` call `parseMaybeForESLint(code, options, { isForESLint: true })`.

Reply:
What about just defensively cloning the options so the user can't see the mutation?

Reply:
You could also do `options = Object.assign({}, options, { parseForESLint: true })`. Or insert `options = Object.create(options)` before this line.

Reply:
It seems strange to have `parseForESLint` be mixed with the user-provided options object. For example, if the user added `parseForESLint: true` to their options object for some reason, they could cause strange behavior.